doc: note on SDK for macOS depends cross-compile

jarolrod opened the pull request:
This PR adds a friendly note for those who are unfamiliar with our depends build system that an extracted SDK is needed before proceeding with a macOS cross-compile. Additionally, it refers a builder to look at [macdeploy](https://github.com/bitcoin/bitcoin/tree/master/contrib/macdeploy#sdk-extraction) for instructions on obtaining the SDK and provides context to where this SDK should reside.

**Master:** [render](https://github.com/bitcoin/bitcoin/blob/master/depends/README.md#for-macos-cross-compilation)
**PR:** [render](https://github.com/bitcoin/bitcoin/blob/241c4017eddbe33a39073766accf556c2cad90a0/depends/README.md#for-macos-cross-compilation)

jarolrod committed:
doc: note on SDK for macOS depends cross-compile

hebasto commented on depends/README.md:
```diff
@@ -49,6 +49,10 @@ The paths are automatically configured and no other options are needed unless ta
 
+Note: You must obtain the macOS SDK before proceeding with a cross-compile.
+The SDK should be placed under `depends/SDKs/`.
+For more information: [macdeploy](../contrib/macdeploy/README.md#sdk-extraction)
```
nit: could it look better
```suggestion
For more information: [SDK Extraction](../contrib/macdeploy/README.md#sdk-extraction)
```
?

jarolrod replied:
addressed in 241c401

hebasto reviewed (approved):
ACK e6c876d364d1a4b71b178ba44e55a562a517aa70

Sjors commented:
ACK

jarolrod commented:
Updated from e6c876d -> 241c401

Changes:
- Changed link for information on SDK extraction from`macdeploy` to `SDK Extraction`

hebasto reviewed (approved):
re-ACK 241c4017eddbe33a39073766accf556c2cad90a0

jonatack commented on depends/README.md:
```diff
@@ -49,6 +49,10 @@ The paths are automatically configured and no other options are needed unless ta
     sudo apt-get install curl librsvg2-bin libtiff-tools bsdmainutils cmake imagemagick libz-dev python3-setuptools libtinfo5 xorriso
 
+Note: You must obtain the macOS SDK before proceeding with a cross-compile.
+The SDK should be placed under `depends/SDKs/`.
```
Does this directory need to be created manually?

jarolrod replied:
Yep, do you think as-is may imply that the directory `SDKs` should be created, or should I explicitly state that.

jonatack replied:
Thanks; I'd say it, so casual users or people who don't look often in `/depends` won't wonder if they are missing something.

jarolrod replied:
adressed in 5e7d1997fce15b3e2fb77e67491237812e1f98d9

jonatack commented on depends/README.md:
```diff
@@ -49,6 +49,10 @@ The paths are automatically configured and no other options are needed unless ta
 
+Note: You must obtain the macOS SDK before proceeding with a cross-compile.
+The SDK should be placed under `depends/SDKs/`.
+For more information: [SDK Extraction](../contrib/macdeploy/README.md#sdk-extraction)
```
nit, I think the presentation with colon and no period at the end would be good if the URL is displayed, or in this case, the rendering might be better with:
```suggestion
For more information, see [SDK Extraction](../contrib/macdeploy/README.md#sdk-extraction).
```

jarolrod replied:
will do, pending your response to: https://github.com/bitcoin/bitcoin/pull/21688#discussion_r615428634

jarolrod replied:
addressed in 5e7d1997fce15b3e2fb77e67491237812e1f98d9

jonatack reviewed (commented):
ACK modulo a nit and a question.

jarolrod commented:
updated from 241c401 -> 5e7d1997fce15b3e2fb77e67491237812e1f98d9

addressed @jonatack suggestions:
- Explicitly state that the `SDKs` directory must be created
- Cleanup presentation of link to SDK extraction instructions

jonatack commented on depends/README.md:
```diff
@@ -49,6 +49,11 @@ The paths are automatically configured and no other options are needed unless ta
+Note: You must obtain the macOS SDK before proceeding with a cross-compile.
+Under the depends directory, create a subdirectory named `SDKs`.
+Then, place the extracted SDK under this new directory.
+For more information, see [SDK Extraction](../contrib/macdeploy/README.md#sdk-extraction).
```
suggested simplication for brevity
```diff
-Note: You must obtain the macOS SDK before proceeding with a cross-compile.
-Under the depends directory, create a subdirectory named `SDKs`.
-Then, place the extracted SDK under this new directory.
-For more information, see [SDK Extraction](../contrib/macdeploy/README.md#sdk-extraction).
+You will need the macOS SDK. In the depends directory, create a subdirectory
+named `SDKs` and place the extracted SDK in it. For more information, see [SDK
+Extraction](../contrib/macdeploy/README.md#sdk-extraction).
```

jonatack commented:
ACK  5e7d1997fce15b3e2fb77e67491237812e1f98d9

One optional suggested comment to shorten/simplify.

hebasto reviewed (approved):
re-ACK 5e7d1997fce15b3e2fb77e67491237812e1f98d9

Sjors commented:
ACK 5e7d199
